### ISSUE TITLE
[IntelliJ][ReportedByUser][ReaderAcct] Azure Plugin for IntelliJ Error - submit with cluster without permission get error

### ISSUE THREAD

**Issue by jingyanjingyan** [Microsoft/azure-tools-for-java#3007]
Build: Dev 1081

Repro Steps:
1. Sign in Reader account
2. Create a new HDInsight config, select one cluster that do not have permission, save
The storage type use default value "Interactive" here.
3. Submit remotly

Result:
Errors
![image](https://user-images.githubusercontent.com/36756024/56009543-a5228780-5ccf-11e9-9fa3-ce538f1c4f4e.png)

IntelliJ build version: 2019.1.1 IC-191.6707.31
OS: Windows 10
JDK: JetBrains s.r.o 1.8.0_202-release
Plugin version: 3.20.0-2019.1
Error message: 
```
java.lang.NullPointerException
	at com.microsoft.azure.hdinsight.spark.run.SparkBatchJobRemoteProcess.lambda$prepareArtifact$10(SparkBatchJobRemoteProcess.java:228)
	at rx.internal.operators.OperatorOnErrorResumeNextViaFunction$4.onError(OperatorOnErrorResumeNextViaFunction.java:140)
	at rx.internal.operators.SingleLiftObservableOperator$WrapSubscriberIntoSingle.onError(SingleLiftObservableOperator.java:81)
	at rx.internal.operators.SingleOnSubscribeMap$MapSubscriber.onError(SingleOnSubscribeMap.java:85)
	at com.microsoft.azure.hdinsight.spark.jobs.JobUtils.lambda$deployArtifact$11(JobUtils.java:688)
	at rx.Single.subscribe(Single.java:1979)
	at rx.internal.operators.SingleOnSubscribeMap.call(SingleOnSubscribeMap.java:45)
	at rx.internal.operators.SingleOnSubscribeMap.call(SingleOnSubscribeMap.java:30)
	at rx.internal.operators.SingleToObservable.call(SingleToObservable.java:39)
	at rx.internal.operators.SingleToObservable.call(SingleToObservable.java:27)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorSubscrib



java.lang.NullPointerException
	at com.microsoft.azure.hdinsight.spark.run.SparkBatchJobRemoteProcess.lambda$prepareArtifact$10(SparkBatchJobRemoteProcess.java:228)
	at rx.internal.operators.OperatorOnErrorResumeNextViaFunction$4.onError(OperatorOnErrorResumeNextViaFunction.java:140)
	at rx.internal.operators.SingleLiftObservableOperator$WrapSubscriberIntoSingle.onError(SingleLiftObservableOperator.java:81)
	at rx.internal.operators.SingleOnSubscribeMap$MapSubscriber.onError(SingleOnSubscribeMap.java:85)
	at com.microsoft.azure.hdinsight.spark.jobs.JobUtils.lambda$deployArtifact$11(JobUtils.java:688)
	at rx.Single.subscribe(Single.java:1979)
	at rx.internal.operators.SingleOnSubscribeMap.call(SingleOnSubscribeMap.java:45)
	at rx.internal.operators.SingleOnSubscribeMap.call(SingleOnSubscribeMap.java:30)
	at rx.internal.operators.SingleToObservable.call(SingleToObservable.java:39)
	at rx.internal.operators.SingleToObservable.call(SingleToObservable.java:27)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:100)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	at rx.internal.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:107)
	at com.microsoft.intellij.rxjava.IdeaSchedulers$1.run(IdeaSchedulers.java:51)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:731)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:586)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:86)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.run(CoreProgressManager.java:403)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:311)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by jingyanjingyan** [comment]
With build dev 1085
Reader account, normal(unlink) gen2 cluster

Submit with interactive storage type:
The same Exception as above

Submit with gen2 storage type:
![image](https://user-images.githubusercontent.com/36756024/56117195-2614a400-5f9a-11e9-8ebd-9affb06669cf.png)

**Comment by t-rufang** [comment]
Fixed in #3040
